### PR TITLE
Support varying colors in a PatchCollection

### PR DESCRIPTION
**Pull request by eric-wieser**
Previously all items would take the color of the first item.

image

### CODE REVIEW

File: tikzplotlib/_patch.py [hunk 1]
```diff
@@ -67,31 +67,27 @@ def draw_patchcollection(data, obj):
...
-        ls = obj.get_linestyle()[0]
-    except IndexError:
-        ls = None
 
-    try:
-        w = obj.get_linewidth()[0]
-    except IndexError:
-        w = None
+    def _or_None(x):
+        if len(x) == 0:
```
Comment (nschloe) [inline tikzplotlib/_patch.py]: how about `return [None] if len(x) == 0 else x`?

Comment (eric-wieser) [reply tikzplotlib/_patch.py]: Done

**Comment by eric-wieser** [comment]
Update also fixes the cmap coloring for `PatchCollection`, which was working in my setup but not your tests because I was calling `plt.show()` first, I think.

File: tikzplotlib/_patch.py (hunk 1 above)
Comment (nschloe) [reply tikzplotlib/_patch.py]: Forgot to push?

Comment (eric-wieser) [reply tikzplotlib/_patch.py]: Pushed to the wrong branch, whoops

**Comment by nschloe** [comment]
Looks like the reference files need another refresh.

**Comment by eric-wieser** [comment]
Same problem as last time - the last digits of the reference output are different on my machine, and I have to remember not to commit them

**Comment by nschloe** [comment]
> Same problem as last time - the last digits of the reference output are different on my machine, and I have to remember not to commit them

I filed a bug for that on https://github.com/nschloe/tikzplotlib/issues/368. If this here turns out to be fiddly for you, I'll cook up something tomorrow that we can merge before this.

**Comment by nschloe** [comment]
Tests now use only 8 digits, so hopefully this should help everyone get reproducible tests on any machine.

**Pull request merged**